File: 2864. Maximum Odd Binary Number/maximumOddBinaryNumberPartitioning.js
/**
 * @param {string} s
 * @return {string}
 */
const maximumOddBinaryNumber = (s) => {
  let left = 0;
  let str = s.split('');

  for (let i = 0; i < str.length; i++) {
    if (s[i] === '1') {
      [str[i], str[left]] = [str[left], str[i]]
      left++;
    }
  }

  [str[left - 1], str[str.length - 1]] = [str[str.length - 1], str[left - 1]]

  return str.join('');
};

const s = "0101"
console.log('maximumOddBinaryNumber: ', maximumOddBinaryNumber(s))

File: 2864. Maximum Odd Binary Number/maximumOddBinaryNumberPartitioning.py
class Solution:
  def maximumOddBinaryNumber(self, s: str) -> str:
    res = [char for char in s]
    left = 0

    for i in range(len(s)):
      if res[i] == '1':
        res[left], res[i] = res[i], res[left]
        left += 1

    res[left - 1], res[len(s) - 1] = res[len(s) - 1], res[left - 1]

    return ''.join(res)
  
solution = Solution()
s = "0101"
print('maximumOddBinaryNumber: ', solution.maximumOddBinaryNumber(s))